examples/cli/src/test/java/io/unitycatalog/cli/access/CliAccessControlFunctionCrudTest.java

```diff
@@ -0,0 +1,210 @@
+package io.unitycatalog.cli.access;
+
+import static io.unitycatalog.cli.access.Step.Expect.FAIL;
+import static io.unitycatalog.cli.access.Step.Expect.SUCCEED;
+
+import io.unitycatalog.cli.access.Step.CommandStep;
+import io.unitycatalog.cli.access.Step.TokenStep;
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+public class CliAccessControlFunctionCrudTest extends CliAccessControlBaseCrudTest {
+
+  List<Step> functionSteps =
+      new ArrayList<>() {
+        {
+          // create a user (principal-1)
```

Comment:
One thing you may want to do is to create an ACL test util that contains all of the steps for catalog/schema permission grants/creation for the securable ACL tests (functions, model, table, volume).   That way, you don't have to copy the same steps over and over for catalog/schema setup.

Reply:
Started down that road.